src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -296,6 +296,52 @@ private static ParameterListSyntax AsParameterList(IEnumerable<SyntaxNode> param
                 SyntaxFactory.AccessorList(SyntaxFactory.List(accessors)));
         }
 
+        public override SyntaxNode GetAccessorDeclaration(Accessibility accessibility, IEnumerable<SyntaxNode> statements)
+            => AccessorDeclaratation(SyntaxKind.GetAccessorDeclaration, accessibility, statements);
+
+        public override SyntaxNode SetAccessorDeclaration(Accessibility accessibility, IEnumerable<SyntaxNode> statements)
+            => AccessorDeclaratation(SyntaxKind.SetAccessorDeclaration, accessibility, statements);
+
+        private static SyntaxNode AccessorDeclaratation(
+            SyntaxKind kind, Accessibility accessibility, IEnumerable<SyntaxNode> statements)
+        {
+            var accessor = SyntaxFactory.AccessorDeclaration(kind);
+            accessor = accessor.WithModifiers(
+                AsModifierList(accessibility, DeclarationModifiers.None, SyntaxKind.PropertyDeclaration));
+
+            accessor = statements == null
+                ? accessor.WithSemicolonToken(SyntaxFactory.Token(SyntaxKind.SemicolonToken))
+                : accessor.WithBody(CreateBlock(statements));
+
+            return accessor;
+        }
+
+        public override SyntaxNode WithAccessorDeclarations(SyntaxNode declaration, IEnumerable<SyntaxNode> accessorDeclarations)
+        {
+            switch (declaration)
+            {
+                case PropertyDeclarationSyntax property:
```

Comment:
:bulb: Would be good to handle events too

❓ Why not throw if called with an unsupported syntax type?